### PR TITLE
dtj chapter 13

### PR DESCRIPTION
**Pull request by thatguy59**

### CODE REVIEW

File: chapter13/dtj/ex3 [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  chdir $directory or die "$! error when trying to open $directory";
+} else {
+  print "Only whitespace supplied, attempting home directory\n";
+  chdir or die "$! error when trying to open home directory";
+}
+
+opendir DIRECTORY, "." or die "Could change to the directory but couldb't open the files";
+foreach my $file (sort readdir DIRECTORY) {
+  print $file . "\n";
+}
```
Comment (codavex) [inline chapter13/dtj/ex3:19]: Would be nicer to close DIRECTORY at the end, I think

File: chapter13/dtj/ex4 [hunk 2]
```diff
@@ -0,0 +1,11 @@
+#!/usr/bin/perl
+
+use strict;
+use warnings;
+
+print "Removing everything from /.....(just kidding)\n";
```
Comment (codavex) [inline chapter13/dtj/ex4:6]: Eeeek! .... Phew!

File: chapter13/dtj/ex7 [hunk 3]
```diff
@@ -0,0 +1,19 @@
+#!/usr/bin/perl
+
+use strict;
+use warnings;
+use Getopt::Std;
+use File::Basename;
+
+our($opt_s);
+getopts('s');
+my $link_function = $opt_s ? \&CORE::symlink : \&CORE::link;
```
Comment (codavex) [inline chapter13/dtj/ex7:10]: Funky

File: chapter13/dtj/ex7 [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+die "Need 2 arguments, source and destination" unless @ARGV == 2;
```
Comment (codavex) [inline chapter13/dtj/ex7:11]: Does the ARGV check here work with the getopts earlier? i.e. does getops remove the -s from ARGV if its found?

File: chapter13/dtj/ex8 [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+if ($directory =~ /\S/ ) {
+  chdir $directory or die "$! error when trying to open $directory";
+} else {
+  print "Only whitespace supplied, attempting home directory\n";
+  chdir or die "$! error when trying to open home directory";
+}
+
+foreach my $file (sort <.* *>) {
+  my $dest = readlink $file;
+  print "$file -> $dest\n" if (-l $file);
```
Comment (codavex) [inline chapter13/dtj/ex8:18]: $dest should be undef from readlink if its is not a link, might be a more efficient test here than -l

Then again, might not.

File: chapter13/dtj/ex7 (hunk 4 above)
Comment (thatguy59) [inline chapter13/dtj/ex7:11]: I did test this and it _seems_ to, not quite sure what would happen if you gave it a switch that it didn't support. Then again, sometimes we don't always know what modifies `@ARGV` so perhaps I should have been more careful.

**Pull request merged**